bundles/org.openhab.binding.argoclima/src/main/java/org/openhab/binding/argoclima/internal/configuration/ArgoClimaConfigurationLocal.java

```diff
@@ -0,0 +1,222 @@
+     * @throws ArgoConfigurationException if at least one of the {@code stubServerListenAddresses} is a hostname and
+     *             cannot be resolved to an IP address
+     */
+    public Set<InetAddress> getStubServerListenAddresses() throws ArgoConfigurationException {
+        var addresses = new LinkedHashSet<InetAddress>();
+        for (var t : stubServerListenAddresses) {
+            try {
+                addresses.add(Objects.requireNonNull(InetAddress.getByName(t)));
+            } catch (UnknownHostException e) {
+                throw ArgoConfigurationException.forInvalidParamValue(
+                        ArgoClimaBindingConstants.PARAMETER_STUB_SERVER_LISTEN_ADDRESSES, t, i18nProvider, e);
+            }
+        }
+        return addresses;
+    }
+
+    @Override
+    public int getRefreshInterval() {
+        if (!this.useLocalConnection) {
+            return 0;
+        }
+        return super.getRefreshInterval();
+    }
+
+    /**
+     * Returns information whether the device-side incoming passwords are to be shown as properties (and if so: in the
+     * clear or replaced with ***)
+     *
+     * @return Configured value
+     */
+    public DeviceSidePasswordDisplayMode getIncludeDeviceSidePasswordsInProperties() {
+        return this.includeDeviceSidePasswordsInProperties;
+    }
+
+    /**
+     * Should the incoming (intercepted) device-side updates be a strict match to local IP (if provided) or hostname
+     * (fallback)
+     *
+     * @return True - if requiring exact match, False - if IP mismatch is allowed
+     */
+    public boolean getMatchAnyIncomingDeviceIp() {
+        return this.matchAnyIncomingDeviceIp;
+    }
+
+    @Override
+    protected String getExtraFieldDescription() {
+        return String.format(
+                "hostname=%s, localDeviceIP=%s, hvacListenPort=%d, connectionMode=%s, useLocalConnection=%s, stubServerPort=%d, stubServerListenAddresses=%s, includeDeviceSidePasswordsInProperties=%s, matchAnyIncomingDeviceIp=%s",
+                getOrDefault(this::getHostname), getOrDefault(this::getLocalDeviceIP), hvacListenPort, connectionMode,
+                useLocalConnection, stubServerPort, getOrDefault(this::getStubServerListenAddresses),
+                includeDeviceSidePasswordsInProperties, matchAnyIncomingDeviceIp);
+    }
+
+    @Override
+    protected void validateInternal() throws ArgoConfigurationException {
+        if (hostname.isEmpty()) {
+            throw ArgoConfigurationException.forEmptyRequiredParam(ArgoClimaBindingConstants.PARAMETER_HOSTNAME,
+                    i18nProvider);
+        }
+
+        if (!useLocalConnection && connectionMode == ConnectionMode.LOCAL_CONNECTION) {
+            throw ArgoConfigurationException.forConflictingParams(
+                    ArgoClimaBindingConstants.PARAMETER_USE_LOCAL_CONNECTION, "OFF",
+                    ArgoClimaBindingConstants.PARAMETER_CONNECTION_MODE, ConnectionMode.LOCAL_CONNECTION, i18nProvider);
```

Comment:
You can get rid if the dependency of the i18nprovider in the configuration classes. The configuration class should not be responsible for translation. only for holding the configuration and logic for validation of it.

If you throw ArgoConfigurationException("specific i18n key for translation") the translation can be performed at the last possible place (when setting the thingstatus later on), see other comments.
that would also make the exception throwing simpler.

More occurences in this class.

Reply:
There's no translation going on (yet). It is merely passing-on the provider (so that at ANY catch side, you can get a localized message for the same if you need). I actually consider this superior to a class just holding a "translation key" but the user having to get some "external" provider from "somewhere" to use the key. 
The way I have it, the exception is self-contained, not encumbering the configuration class.

Reply:
As there are also benefits, we could leave it as is. The attempts i made locally did not necesairly improve and ended up in the same but different. So let's move on.